eclipse-scout-core/src/testing/form/fields/smartfield/proposalFieldSpecHelper.ts

```diff
@@ -59,20 +59,34 @@ export const proposalFieldSpecHelper = {
       popup.doneAction.doAction();
     } else {
       field.$field.val(text);
-      field.acceptInput();
+      await field.acceptInput();
     }
   },
 
-  async selectLookupRow(field: ProposalField, text: string): Promise<LookupRow<string>> {
+  async selectLookupRow(field: ProposalField, text: string, strategy = LookupRowSelectionStrategy.CLICK): Promise<LookupRow<string>> {
     // find row for text
     const popup = await proposalFieldSpecHelper.openPopup(field);
+    const smartField = field.touchMode ? (popup as SpecSmartFieldTouchPopup<string>)._field : (popup as SmartFieldPopup<string>).smartField;
     const proposalChooser = field.touchMode ? (popup as SpecSmartFieldTouchPopup<string>)._widget : (popup as SmartFieldPopup<string>).proposalChooser;
-    const table = proposalChooser.content;
+    const table = proposalChooser.content as Table;
     const row = table.rows.find(r => r.cells[0].text === text);
 
-    // trigger row mousedown and mouseup
-    JQueryTesting.triggerMouseDown(row.$row);
-    JQueryTesting.triggerMouseUp(row.$row);
+    switch (strategy) {
+      case LookupRowSelectionStrategy.CLICK:
+        JQueryTesting.triggerMouseDown(row.$row);
+        JQueryTesting.triggerMouseUp(row.$row);
+        break;
+      case LookupRowSelectionStrategy.ENTER:
+        table.selectRow(row);
+        JQueryTesting.triggerKeyDownCapture(smartField.$field, keys.ENTER);
+        break;
+      case LookupRowSelectionStrategy.EXACT_TEXT:
+        smartField.$field.val(text);
+        let popupDeffered = proposalFieldSpecHelper.getDeferredForEvent(popup, 'close');
```

Comment:
```suggestion+0-0
        let popupDeferred = proposalFieldSpecHelper.getDeferredForEvent(popup, 'close');
```

Also this gives an eslint-error. Please add curly braces.